.pre-commit-config.yaml

```diff
@@ -0,0 +1,22 @@
+# See https://pre-commit.com for more information
+# See https://pre-commit.com/hooks.html for more hooks
+repos:
+-   repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v3.2.0
+    hooks:
+    -   id: trailing-whitespace
+    -   id: end-of-file-fixer
+    -   id: check-yaml
+    -   id: check-added-large-files
+-   repo: https://github.com/golangci/golangci-lint
+    rev: v1.58.1
+    hooks:
+    -   id: golangci-lint
+- repo: local
+  hooks:
+    - id: go-test-short
+      name: go-test-short
+      entry: bash -c 'go test -tags conn_insecure -short ./...'
```

Comment:
You run tests on each commit?

Reply:
every commit should compile lest we can't `git bissect` on it.

Reply:
sure but we don't merge things which don't pass the full build, and we squash

Reply:
I guess it'd be best to change it to do that on push only yeah.